dbms/src/Flash/Statistics/ExchangeReceiverImpl.cpp

```diff
@@ -48,15 +48,16 @@ void ExchangeReceiverStatistics::collectExtraRuntimeDetail()
     {
         for (const auto & io_stream : it->second)
         {
-            auto * exchange_receiver_stream = dynamic_cast<ExchangeReceiverInputStream *>(io_stream.get());
             /// InBoundIOInputStream of ExchangeReceiver should be ExchangeReceiverInputStream
-            assert(exchange_receiver_stream);
-            const auto & connection_profile_infos = exchange_receiver_stream->getConnectionProfileInfos();
-            assert(connection_profile_infos.size() == partition_num);
-            for (size_t i = 0; i < partition_num; ++i)
+            if (auto * exchange_receiver_stream = dynamic_cast<ExchangeReceiverInputStream *>(io_stream.get()); exchange_receiver_stream)
```

Comment:
when will `exchange_receiver_stream` be nullptr?

Reply:
Currently `exchange_receiver_stream` must not be nullptr.
This is just to avoid a tiflash crash in any case.